php-classes/Slate/CBL/Demonstrations/Demonstration.php

```diff
@@ -112,10 +112,17 @@ public function getCompetencyCompletions()
 
         $completions = [];
         foreach ($competencies AS $Competency) {
-            $completion = $Competency->getCompletionForStudent($this->Student);
-            $completion['StudentID'] = $this->StudentID;
-            $completion['CompetencyID'] = $Competency->ID;
-            $completions[] = $completion;
+            $StudentCompetency = StudentCompetency::getCurrentForStudent($this->Student);
+
+            $completions[] = [
```

Comment:
@nbey let's add a method `getCompletion` to `StudentCompetency` that handles returning this data structure since it's a standard of sorts and we're repeating it in a bunch of places.